pandas/core/generic.py

```diff
@@ -5906,6 +5908,129 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
```

Comment:
I'm finding these temporary variables a bit hard to follow. I suspect that both `changeit` and `constructit` can be removed, perhaps at the cost of always constructing a new Series?

Reply:
`changeit` is `True` if you have to change the dtype.  However, if we drop through to where we are converting a `float` array to integer, the result of the `astype()` method call is a `Series` so `changeit` remains False, and `constructit` is False as well, because you already have a new `Series`. 

But I see your point.  We can always call `astype()` as it will return a new Series even if the dtype didn't change.

Reply:
cleaned up